src/main/java/pl/edu/mimuw/logic/Formula.java

```diff
@@ -1,8 +1,12 @@
 package pl.edu.mimuw.logic;
 
+import java.util.ArrayList;
+
 public interface Formula {
+  ArrayList<String> getAllVariables();
```

Comment:
lepiej by było jednak `Set<String>` by wyrazic, ze elementy nie beda sie powtarzac
a jak juz atk chcesz z lista to `List<String>` używajac typu interfejsu